consensus/babe/types.go

```diff
@@ -43,10 +49,63 @@ type AuthorityData struct {
 	weight uint64
 }
 
-//nolint:structcheck
-type BlockInherentsData struct {
-	timstap0 int64 //nolint:unused
-	babeslot int64 //nolint:unused
+var Timstap0 = []byte("timstap0")
+var Babeslot = []byte("babeslot")
+
+// InherentsData contains a mapping of inherent keys to values
+// Keys must be 8 bytes, values are a variable-length byte array
+type InherentsData struct {
+	data map[[8]byte]([]byte)
+}
+
+func NewInherentsData() *InherentsData {
+	return &InherentsData{
+		data: make(map[[8]byte]([]byte)),
+	}
+}
+
+func (d *InherentsData) SetInt64Inherent(key []byte, data uint64) error {
+	if len(key) != 8 {
+		return errors.New("inherent key must be 8 bytes")
```

Comment:
Should we update the `key` input type instead of checking its length?

Reply:
the issue is we declare the keys as global vars, and we can't turn a string into a fixed byte array:
eg. right now we have this, which is fine: `var Timstap0 = []byte("timstap0")`
but this doesn't work: `var Timstap0 = [8]byte("timstap0")`